content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,695 @@
+
+Like the Prisoner Dilemma, the execution of Glacier on a single
+proposal doesn't have very complicated game theoretics.  Simply
+conceived, multiple runs of Glacier could track a linear chain of
+consensus like any other contemporary implementation. But in practice,
+leaderless consensus may be used contribute finalization of multiple
+proposals arranged in a directed acyclic graph.  Nodes would naturally
+choose to compute the consensus of directed graph of the proposals
+that they are interested in.  This autosharding property superceeds
+any possible Nakomota consensus mechanism, as any proof of work
+mechanism would by defition need to increase its the work necessary
+for its continued constant security of all chains of interest.
+
+# Glacier Algorithm Specification
+
+The Glacier consensus algorithm computes a boolean decision on a
+proposition via a set of distributed computational nodes.  Glacier is
+a leaderless probabilistic binary consensus algorithm with fast
+finality that provides good reliability for network and Byzantine
+fault tolerance.
+
+## Algorithm 
+
+The algorithm begins with considering the truth status of a given
+proposition.
+
+A proposal is formulated to which consensus of truth or falsity is
+desired.  Each node that participates starts the protocol with an
+opinion on the proposal, represented in the sequel as `no`, `none`,
+and `yes`.
+
+A new proposition is discovered either by local creation or in
+response to a query, a node checks its local opinion.  If the node can
+compute a justification of the proposal, it sets its opinion to one of
+`yes` or `no`.  If it cannot form an opinion, it leaves its opinion as
+`none`.
+
+The node then participates in a number of query rounds in which it
+solicits other node's opinion in query rounds.  Given a set of `n`
+leaderless computational nodes, a gossip-based protocol is presumed to
+exist which allows members to discover, join, and leave a weakly
+transitory maximally connected graph.  Joining this graph allows each
+node to view a possibly incomplete node membership list of all other
+nodes.  This view may change as the protocol advances, as nodes join
+and leave.
+
+
+### Proposal Identification
+
+The node has a semantics and serialization of the proposal, of which
+it sets an initial opinion:
+
+     opinion
+        <-- initial opinion on truth of the proposal
+            as one of: {NO, NONE, YES}
+    
+The proposal proceeds in asynchronous rounds, in which each node
+queries `k` randomly sampled nodes for their opinions until a decision
+about local finality is achieved.  The liveness of the algorithm is
+severely constrained in the absence of timeouts for a round to
+proceed.  When a given node has finalized its decision on the
+proposal, it enters a quiescent state in which it optionally discards
+all information gathered during the query process retaining only the
+final opinion on the truth of the proposal.
+
+### Setup Parameters
```

Comment:
@corpetty I could use a little more instruction on how to revise: 

1.  A Greek identifier for each constant/parameter assigination?  LaTeX is still a little wonky, so I just did all the actual algorithm in straight Markdown.  I would use UTF-8 glyphs at the moment for this labeling.

2.  By "formal" do you mean some sort of type specification?  I eliminated the use of floats on purpose, so that currently all one needs is positive integer ratios.  We could put an upper limit on the size of the integers, but I don't have justification for this other than "what would reasonably fit" efficiently on contemporary architectures.

2.

Reply:
On page 7 of  https://www.overleaf.com/read/vrqshkdxkfxz I have written mathematical description of the Glacier algorithm  and perhaps some of the notation introduced  can be also used in this document.